### PR TITLE
front: tests: setup playwright without poetry and make them fully parallel

### LINKED ISSUE #5212: Refactor E2E tests : config-playwright without python
### Description and goal

Actually, playwright e2e-tests uses pytest configuration, we need to set the playwright config fully independent

### Acceptance criteria

- [x] `e2e-tests` script only asks for playwright config in package-json
- [x] tests can be launch
- [x] tests can be launch parallelized
- [ ] test_e2e.py is removed as it is at last useless

### LINKED ISSUE #5252: Refactor E2E tests : home-page, project-creation, study-creation, scenario-creation
# Description

Refactor operationnal studies, stdcm page, pathfinding e2e tests according to what's been defined in #5022

# Goal

- [x] add apiContext in each test to make them parallelisable
- [x] add missing tests fore each user action

# Acceptance criteria

The following e2e have been refectored :
- [x] project management
- [x] study management
- [x] scenario management

### PR DESCRIPTION
**Pull request by Yohh**
- edit playwright.config.ts to set parallel option and remove webkit project
- edit package.json
- add uuidv4 to make names unique in tests
- add apiContext in tests to make api calls and make every test independent
- isolate api calls in utils.ts
- add BASE_URL in testVariables.ts

close #5212 
close #5252

### CODE REVIEW

**Review by bloussou — COMMENTED** [review]
Nice ! Good start. I think setup/cleanup api calls must be done out of tests using test setupteardown methods.
Global setup/teardwon (to load infra for instance) : https://playwright.dev/docs/test-global-setup-teardown
Playwright fixtures to create and delete a project for instance: https://playwright.dev/docs/test-fixtures
Test before/after beforeAll/afterAll (maybe easier than fixtures to setup...) : https://playwright.dev/docs/api-testing#setup-and-teardown

File: front/tests/002-project-management.spec.ts [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    name: `${project.name} ${uuidv4()}`,
+    budget: 1234567890,
+  });
+});
+
+test.afterEach(async () => {
+  await deleteApiRequest(`/projects/${newProjectData.id}/`);
+});
+
+test.describe('Test is operationnal study  : project workflow is working properly', () => {
```
Comment (Math-R) [inline front/tests/002-project-management.spec.ts]: test is ? or test if ?

File: front/tests/003-study-management.spec.ts [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+    await studyPage.openStudyModalUpdate();
+
+    await studyPage.clickStudyDeleteConfirmBtn();
+
+    await playwrightHomePage.goToHomePage();
+    await playwrightHomePage.goToOperationalStudiesPage();
+
+    await expect(studyPage.getStudyByName(newStudyData.name)).not.toBeVisible();
+
+    const deleteProject = await deleteApiRequest(`/projects/${newProjectData.id}/`);
```
Comment (Math-R) [inline front/tests/003-study-management.spec.ts]: You already test it in porject-management.spec.ts so you can just use the after each hook

File: front/tests/004-scenario-management.spec.ts [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+    await scenarioPage.clickScenarioDeleteConfirmBtn();
+
+    await playwrightHomePage.goToHomePage();
+    await playwrightHomePage.goToOperationalStudiesPage();
+    await projectPage.openProjectByTestId(newProjectData.name);
+    await studyPage.openStudyByTestId(newStudyData.name);
+
+    await expect(scenarioPage.getScenarioByName(newScenarioData.name!)).not.toBeVisible();
+
+    const deleteProject = await deleteApiRequest(`/projects/${newProjectData.id}/`);
```
Comment (Math-R) [inline front/tests/004-scenario-management.spec.ts]: Same as previously you already tested it in project-management.spec.ts

File: front/tests/assets/utils.ts [hunk 4]
```diff
@@ -77,3 +77,29 @@ export default async function createCompleteScenario(
   await scenarioPage.checkToastSNCFTitle('Train ajouté');
   await scenarioPage.returnSimulationResult();
 }
+
+// API requests
+
+export const getApiRequest = async (url: string) => {
+  const apiContext = await request.newContext({
+    baseURL: BASE_URL,
```
Comment (Math-R) [inline front/tests/assets/utils.ts]: Remember to handle fallback

File: front/tests/assets/utils.ts [hunk 5]
```diff
@@ -77,3 +77,29 @@ export default async function createCompleteScenario(
...
+  });
+  const response = await apiContext.get(url);
+  return response.json();
+};
+
+export const postApiRequest = async <T>(url: string, data: T) => {
+  const apiContext = await request.newContext({
+    baseURL: BASE_URL,
```
Comment (Math-R) [inline front/tests/assets/utils.ts]: same

File: front/tests/assets/utils.ts [hunk 6]
```diff
@@ -77,3 +77,29 @@ export default async function createCompleteScenario(
...
+  });
+  const newProject = await apiContext.post(url, { data });
+  return newProject.json();
+};
+
+export const deleteApiRequest = async (url: string) => {
+  const apiContext = await request.newContext({
+    baseURL: BASE_URL,
```
Comment (Math-R) [inline front/tests/assets/utils.ts]: same

File: front/tests/pages/scenario-page-model.ts [hunk 7]
```diff
@@ -78,6 +86,11 @@ class PlaywrightScenarioPage {
   readonly getSuccessBtn: Locator;
 
   constructor(readonly page: Page) {
+    this.getScenarioUpdateBtn = page.getByTitle('Modifier le scénario');
+    this.getScenarioDeleteConfirmBtn = page
+      .locator('#modal-content')
+      .getByText('Supprimer', { exact: true });
```
Comment (Math-R) [inline front/tests/pages/scenario-page-model.ts:93]: try to locate with data-id

File: front/tests/002-project-management.spec.ts [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+import { PlaywrightHomePage } from './pages/home-page-model';
+import project from './assets/operationStudies/project.json';
+import { ProjectPage } from './pages/project-page-model';
+import PlaywrightCommonPage from './pages/common-page-model';
+import { deleteApiRequest, getApiRequest, postApiRequest } from './assets/utils';
+
+let newProjectData: ProjectResult;
+
+test.beforeEach(async () => {
+  newProjectData = await postApiRequest('/projects/', {
```
Comment (Uriel-Sautron) [inline front/tests/002-project-management.spec.ts:13]: This code is also in study-management  scenario-management. Maybe we can find a way to make it reusable?

File: front/tests/002-project-management.spec.ts (hunk 1 above)
Comment (Yohh) [reply front/tests/002-project-management.spec.ts]: done

File: front/tests/003-study-management.spec.ts (hunk 2 above)
Comment (Yohh) [reply front/tests/003-study-management.spec.ts]: done

File: front/tests/004-scenario-management.spec.ts (hunk 3 above)
Comment (Yohh) [reply front/tests/004-scenario-management.spec.ts]: done

File: front/tests/assets/utils.ts (hunk 4 above)
Comment (Yohh) [reply front/tests/assets/utils.ts]: done

File: front/tests/assets/utils.ts (hunk 5 above)
Comment (Yohh) [reply front/tests/assets/utils.ts]: done

File: front/tests/assets/utils.ts (hunk 6 above)
Comment (Yohh) [reply front/tests/assets/utils.ts]: done

File: front/tests/pages/scenario-page-model.ts (hunk 7 above)
Comment (Yohh) [reply front/tests/pages/scenario-page-model.ts:93]: done

File: front/tests/002-project-management.spec.ts (hunk 8 above)
Comment (Yohh) [reply front/tests/002-project-management.spec.ts:13]: it should be possible, but those beforeEach and afterEach interact with an external variable newProjectData wich is used in every tests in association with an uuid(), setting those globally may turn the code more complicated than we need

**Review by flomonster — APPROVED** [review]
Reviewed `integraion_tests.yml`.

**Comment by Math-R** [comment]
lgtm =)

**Review by multun — APPROVED** [review]

**Review by Math-R — APPROVED** [review]

**Pull request merged**